src/java.base/share/classes/java/util/AbstractMap.java

```diff
@@ -924,4 +896,21 @@ public String toString() {
         public <T> T[] toArray(T[] a) { return view().toArray(a); }
         public String toString() { return view().toString(); }
     }
+
+    // Iterator implementations.
+    // Here, we avoid polymorphism to ensure full VM optimization capabilities.
```

Comment:
I would remove this comment about avoiding polymorphism, since I think it's a total red herring and is potentially confusing. Otherwise this is fine.